cpp/unreal_plugins/Vehicle/Source/Vehicle/VehiclePawn.cpp

```diff
@@ -53,45 +70,227 @@ AVehiclePawn::AVehiclePawn(const FObjectInitializer& object_initializer) :
+    if (Config::s_initialized_) {
+        camera_location = {
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.LOCATION_X"),
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.LOCATION_Y"),
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.LOCATION_Z")};
+        camera_rotation = {
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.ROTATION_PITCH"),
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.ROTATION_YAW"),
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.ROTATION_ROLL")};
+        field_of_view = Config::get<float>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.FOV");
+        aspect_ratio = Config::get<float>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.ASPECT_RATIO");
+    } else {
+        // OpenBot defaults, see python/spear/config/default_config.vehicle.yaml
+        camera_location = {9.0, 5.0, 10.0};
+        camera_rotation = FRotator::ZeroRotator;
+        field_of_view = 70.0f;
+        aspect_ratio = 1.333333f;
+    }
+
+    CameraComponent = CreateDefaultSubobject<UCameraComponent>(Unreal::toFName("camera_component"));
+    SP_ASSERT(CameraComponent);
+    CameraComponent->SetRelativeLocationAndRotation(camera_location, camera_rotation);
+    CameraComponent->SetupAttachment(GetMesh());
+    CameraComponent->bUsePawnControlRotation = false;
+    CameraComponent->FieldOfView = field_of_view;
+    CameraComponent->AspectRatio = aspect_ratio;
+
+    // UBoxComponent
+    FVector imu_location;
+    FRotator imu_rotation;
+    if (Config::s_initialized_) {
+        imu_location = {
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.LOCATION_X"),
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.LOCATION_Y"),
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.LOCATION_Z")};
+        imu_rotation = {
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.ROTATION_PITCH"),
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.ROTATION_YAW"),
+            Config::get<double>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.ROTATION_ROLL")};
+    } else {
+        // OpenBot defaults, see python/spear/config/default_config.vehicle.yaml
+        imu_location = {8.0, 0.0, 10.0};
+        imu_rotation = FRotator::ZeroRotator;
+    }
+
+    ImuComponent = CreateDefaultSubobject<UBoxComponent>(Unreal::toFName("imu_component"));
+    SP_ASSERT(ImuComponent);
+    ImuComponent->SetRelativeLocationAndRotation(imu_location, imu_rotation);
+    ImuComponent->SetupAttachment(GetMesh());
+
+    // UVehicleMovementComponent
+    MovementComponent = dynamic_cast<UVehicleMovementComponent*>(GetVehicleMovementComponent());
+    SP_ASSERT(MovementComponent);
+
+    // UInputActionComponent
+    input_action_component_ = CreateDefaultSubobject<UInputActionComponent>(Unreal::toFName("input_action_component"));
+    SP_ASSERT(input_action_component_);
```

Comment:
Do we need to call `SetupAttachment(...)` or `RegisterComponent()` here?

Reply:
No, we don't need to call `RegisterComponent()`, but setting up `SetupAttachment()` is a good. Updated with this change.

Reply:
Is `SetupAttachment(...)` only for `USceneComponent`?

If so, we should make a consistent rule throughout the codebase, i.e., always call `RegisterComponent()` immediately after creating it, and then create `SetupAttachment(...) if it is a `USceneComponent`.